source/shared/core_conn.cpp

```diff
@@ -654,6 +650,23 @@ void get_server_version( sqlsrv_conn* conn, char** server_version, SQLSMALLINT&
 // and return the string of the processor name.
 const char* get_processor_arch( void )
 {
+#ifndef _WIN32
```

Comment:
Please double check if Apple also has such system info names. If not, it seems more straightforward to retain #ifdef __linux__
Also, all these strcmp() calls... is testing for equality, shouldn't we use (! strcmp(...))???